frappe/desk/form/load.py

```diff
@@ -201,6 +203,20 @@ def get_versions(doc: "Document") -> list[dict]:
 	)
 
 
+def get_error_log_exists(doc: "Document") -> bool:
+	if has_permission("Error Log"):
+		return frappe.db.exists("Error Log", {"reference_doctype": doc.doctype, "reference_name": doc.name})
```

Comment:
Adding one read on MyISAM table on every document read is a horrible thing for performance.

`reference_name` is not even indexed. 

How the fuck shit like this gets merged?

Reply:
We had a bit of a discussion on this over here: https://github.com/frappe/frappe/pull/27777#discussion_r1760819872

With that discussion in mind, would you still suggest to approach this differently?

> random QoL features

On the basis of my continued production usage of the framework, that's a bit of a mischaracterization, though. It does not take into account the many troubles that users face due to the lack of auditablility when things don't work as intended. The time-to-due-escalation had dropped by a valuable 300% due to this increased visibility. This is a huge deal in avoiding cascading issues that are hard to fix manually.

Reply:
Note: it also _did_ work reasonably well with the client lazily fetching this info, the delay was noticeable, but not of issue. So I could revert to that. But let's consider first how to approach this requirement properly / better.

Reply:
Off Topic: what's the purpose of an index on `creation`? Isn't that useless because it's carnality is almost inevitably close to `count(*)`?

Reply:
> `reference_name` is not even indexed.

`reference_doctype` is probably the better (or good enough) choice to narrow down candidates in the query?

Reply:
> MyISAM has implicit global locking

https://dba.stackexchange.com/questions/54426/should-we-explicitly-lock-myisam-tables-or-is-it-implicit

claims "MyISAM only locks for INSERT, UPDATE, and DELETE (a.k.a. DML)" which seems reasonable.

Are we wary of DML locks, here?

Reply:
Potential resolution via DM:

1. Add a feature flag in sys settings (cached)
2. Disable by default

Reply:
> reference_doctype is probably the better (or: good enough; or: balanced) choice to narrow down candidates in the query?

No. Name has higher cardinality. Doctype hardly narrows it down.

Reply:
> claims "MyISAM only locks for INSERT, UPDATE, and DELETE (a.k.a. DML)" which seems reasonable.

When an error log is being inserted, no one can read the table. When someone is reading the table, no one can write to it. Now if that reader is not even using indexes then you can imagine the slowdown this causes. 

I've seen huge tail latency spikes because of this, it's not reasonable.

Reply:
I understand that we actually agree that "random QoL features" is a bit of an over-characterization what this is.

> slow it down for everyone

As per https://github.com/frappe/frappe/pull/27777#discussion_r1760862326 this has never be the goal and even was the subject of active concern. We just came to a different conclusion for XYZ. That happens, but the thought process itself _was_ sound.